Tests/js/views/services/assets/ez-languageselectionboxviewservice-tests.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-languageselectionboxviewservice-tests', function (Y) {
+    var getViewParametersTest,
+        Mock = Y.Mock, Assert = Y.Assert;
+
+    getViewParametersTest = new Y.Test.Case({
+        name: "eZ Language Selection Box View Service getViewParameters test",
+
+        setUp: function () {
+            this.app = new Mock();
+            this.systemLanguageList = {
+                'eng-EN': {languageCode: 'eng-EN', name: 'EN'},
+                'pol-PL': {languageCode: 'pol-PL', name: 'PL'}
+            };
+
+            Mock.expect(this.app, {
+                method: 'get',
+                args: ['systemLanguageList'],
+                returns: this.systemLanguageList
+            });
+
+            this.service = new Y.eZ.LanguageSelectionBoxViewService({
+                app: this. app
+            });
+        },
+
+        tearDown: function () {
+            this.service.destroy();
+            delete this.service;
+        },
+
+        "Should return the systemLanguageList object": function () {
+            Assert.areEqual(
+                this.systemLanguageList,
+                this.service.getViewParameters().systemLanguageList,
+                "getViewParameters should return the systemLanguageList object"
+            );
+        },
+
+        "Should return the parameters object": function () {
+            var parameters = {
+                    'systemLanguageList': this.systemLanguageList
```

Comment:
that's a very bad choice, since this property is added in the returned object in `_getViewParameters`, this test does not check if the `parameters` object properties are kept or not